### PR TITLE
New formula: telegram-cli

### PR DESCRIPTION
**Pull request by alexreg**
Test not included because any worthwhile test is non-trivial.

### CODE REVIEW

File: Formula/telegram-cli.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class TelegramCli < Formula
+  desc "Command-line interface for Telegram."
+  homepage "https://github.com/vysheng/tg"
+  # Use the tag instead of the tarball to get submodules
+  url "https://github.com/vysheng/tg.git",
+      :tag => "1.3.1",
+      :revision => "5935c97ed05b90015418b5208b7beeca15a6043c"
+  head "https://github.com/vysheng/tg.git"
+  revision 1
```
Comment (MikeMcQuaid) [inline Formula/telegram-cli.rb]: Remove the `revision`

File: Formula/telegram-cli.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  depends_on "pkg-config" => :build
+  depends_on "readline" => :build
+  depends_on "libevent" => :build
+  depends_on "openssl" => :build
+  depends_on "libconfig" => :recommended
+  depends_on "jansson" => :recommended
+  depends_on "lua" => :recommended
+  depends_on "python" => :recommended
```
Comment (MikeMcQuaid) [inline Formula/telegram-cli.rb]: It's probably just worth leaving these all on by default.

File: Formula/telegram-cli.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  url "https://github.com/vysheng/tg.git",
+      :tag => "1.3.1",
+      :revision => "5935c97ed05b90015418b5208b7beeca15a6043c"
+  head "https://github.com/vysheng/tg.git"
+  revision 1
+
+  depends_on "pkg-config" => :build
+  depends_on "readline" => :build
+  depends_on "libevent" => :build
+  depends_on "openssl" => :build
```
Comment (MikeMcQuaid) [inline Formula/telegram-cli.rb]: Pretty sure all these libraries are needed at runtime.

File: Formula/telegram-cli.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  depends_on "libconfig" => :recommended
+  depends_on "jansson" => :recommended
+  depends_on "lua" => :recommended
+  depends_on "python" => :recommended
+
+  patch :DATA
```
Comment (MikeMcQuaid) [inline Formula/telegram-cli.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Formula/telegram-cli.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  def install
+    args = %W[
+      --prefix=#{prefix}
+      CFLAGS=-I#{Formula["readline"].include}
+      CPPFLAGS=-I#{Formula["readline"].include}
+      LDFLAGS=-L#{Formula["readline"].lib}
```
Comment (MikeMcQuaid) [inline Formula/telegram-cli.rb]: Are you sure these arguments are needed? Seems like `configure` should be adding them instead.

File: Formula/telegram-cli.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    args << "--disable-libconfig" if build.without? "libconfig"
+    args << "--disable-json" if build.without? "jansson"
+    args << "--disable-liblua" if build.without? "lua"
+    args << "--disable-python" if build.without? "python"
+
+    system "./configure", *args
+    system "make"
+
+    bin.install "bin/telegram-cli"
```
Comment (MikeMcQuaid) [inline Formula/telegram-cli.rb]: Please request a `make install` from upstream if there isn't one already.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
> Test not included because any worthwhile test is non-trivial.

We need a test of some form, even if trivial.

**Comment by alexreg** [comment]
Sure. What about just doing “telegram-cli -h”?

**Comment by MikeMcQuaid** [comment]
@alexreg Yep, better than nothing, thanks.

**Comment by alexreg** [comment]
> On 29 Jan 2017, at 17:01, Mike McQuaid <notifications@github.com> wrote:
> 
> @MikeMcQuaid requested changes on this pull request.
> 
> In Formula/telegram-cli.rb <https://github.com/Homebrew/homebrew-core/pull/9400#pullrequestreview-18997076>:
> 
> > @@ -0,0 +1,57 @@
> +class TelegramCli < Formula
> +  desc "Command-line interface for Telegram."
> +  homepage "https://github.com/vysheng/tg"
> +  # Use the tag instead of the tarball to get submodules
> +  url "https://github.com/vysheng/tg.git",
> +      :tag => "1.3.1",
> +      :revision => "5935c97ed05b90015418b5208b7beeca15a6043c"
> +  head "https://github.com/vysheng/tg.git"
> +  revision 1
> Remove the revision
> 
Okay, done.

> 
> In Formula/telegram-cli.rb <https://github.com/Homebrew/homebrew-core/pull/9400#pullrequestreview-18997076>:
> 
> > +  homepage "https://github.com/vysheng/tg"
> +  # Use the tag instead of the tarball to get submodules
> +  url "https://github.com/vysheng/tg.git",
> +      :tag => "1.3.1",
> +      :revision => "5935c97ed05b90015418b5208b7beeca15a6043c"
> +  head "https://github.com/vysheng/tg.git"
> +  revision 1
> +
> +  depends_on "pkg-config" => :build
> +  depends_on "readline" => :build
> +  depends_on "libevent" => :build
> +  depends_on "openssl" => :build
> +  depends_on "libconfig" => :recommended
> +  depends_on "jansson" => :recommended
> +  depends_on "lua" => :recommended
> +  depends_on "python" => :recommended
> It's probably just worth leaving these all on by default.
> 
It’s nice to have the option though, isn’t it? I mean, they’re already on by default (they should be —without-* options). Maybe libconfig I can make essential, if you prefer.

> 
> In Formula/telegram-cli.rb <https://github.com/Homebrew/homebrew-core/pull/9400#pullrequestreview-18997076>:
> 
> > @@ -0,0 +1,57 @@
> +class TelegramCli < Formula
> +  desc "Command-line interface for Telegram."
> +  homepage "https://github.com/vysheng/tg"
> +  # Use the tag instead of the tarball to get submodules
> +  url "https://github.com/vysheng/tg.git",
> +      :tag => "1.3.1",
> +      :revision => "5935c97ed05b90015418b5208b7beeca15a6043c"
> +  head "https://github.com/vysheng/tg.git"
> +  revision 1
> +
> +  depends_on "pkg-config" => :build
> +  depends_on "readline" => :build
> +  depends_on "libevent" => :build
> +  depends_on "openssl" => :build
> Pretty sure all these libraries are needed at runtime.
> 
Except pkg-config, yes, you’re right! I had assumed they were getting linked statically for some reason.

> 
> In Formula/telegram-cli.rb <https://github.com/Homebrew/homebrew-core/pull/9400#pullrequestreview-18997076>:
> 
> > +  url "https://github.com/vysheng/tg.git",
> +      :tag => "1.3.1",
> +      :revision => "5935c97ed05b90015418b5208b7beeca15a6043c"
> +  head "https://github.com/vysheng/tg.git"
> +  revision 1
> +
> +  depends_on "pkg-config" => :build
> +  depends_on "readline" => :build
> +  depends_on "libevent" => :build
> +  depends_on "openssl" => :build
> +  depends_on "libconfig" => :recommended
> +  depends_on "jansson" => :recommended
> +  depends_on "lua" => :recommended
> +  depends_on "python" => :recommended
> +
> +  patch :DATA
> Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!
> 
Already done. I explained this in a commit message, but not the pull-request, sorry. We’ll keep the patch for now, then remove it when/if it’s accepted upstream, if that’s okay?

> 
> In Formula/telegram-cli.rb <https://github.com/Homebrew/homebrew-core/pull/9400#pullrequestreview-18997076>:
> 
> > +  depends_on "readline" => :build
> +  depends_on "libevent" => :build
> +  depends_on "openssl" => :build
> +  depends_on "libconfig" => :recommended
> +  depends_on "jansson" => :recommended
> +  depends_on "lua" => :recommended
> +  depends_on "python" => :recommended
> +
> +  patch :DATA
> +
> +  def install
> +    args = %W[
> +      --prefix=#{prefix}
> +      CFLAGS=-I#{Formula["readline"].include}
> +      CPPFLAGS=-I#{Formula["readline"].include}
> +      LDFLAGS=-L#{Formula["readline"].lib}
> Are you sure these arguments are needed? Seems like configure should be adding them instead.
> 
They are I’m afraid. The `configure` script is doing something weird, as @tdsmith and I diagnosed yesterday on IRC. If they’re not included, then it picks up the system *readline* library, which is a problem, because it lacks some required feature.

> 
> In Formula/telegram-cli.rb <https://github.com/Homebrew/homebrew-core/pull/9400#pullrequestreview-18997076>:
> 
> > +    args = %W[
> +      --prefix=#{prefix}
> +      CFLAGS=-I#{Formula["readline"].include}
> +      CPPFLAGS=-I#{Formula["readline"].include}
> +      LDFLAGS=-L#{Formula["readline"].lib}
> +    ]
> +
> +    args << "--disable-libconfig" if build.without? "libconfig"
> +    args << "--disable-json" if build.without? "jansson"
> +    args << "--disable-liblua" if build.without? "lua"
> +    args << "--disable-python" if build.without? "python"
> +
> +    system "./configure", *args
> +    system "make"
> +
> +    bin.install "bin/telegram-cli"
> Please request a make install from upstream if there isn't one already.
> 
This is done now. :)

**Pull request closed without merging**